tests/TestUtils/StrTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Tests\TestUtils;
+
+use PHPUnit\Framework\TestCase;
+
+use function implode;
+
+class StrTest extends TestCase
+{
+    // describe('dedent')
+
+    /**
+     * @see it('removes indentation in typical usage')
+     */
+    public function testRemovesIndentationInTypicalUsage(): void
+    {
+        $output = Str::dedent('
+          type Query {
+            me: User
+          }
+
+          type User {
+            id: ID
+            name: String
+          }
+        ');
+        self::assertEquals(
+            implode(
```

Comment:
Wouldn't it be easier to use a NOWDOC?

Reply:
I was thinking about it for a short moment myself, but I dismissed the idea. I'm trying to keep the code generally and tests particularly as much close to the reference implementation as possible to make it easy to reflect future changes.

`dedent()` deals with whitespaces and empty lines, so it's important to be clear about what the testing string is. I believe using `implode()` is more explicit than NOWDOC there is easier to get lost in the indentation etc. It also makes changes like this one super obvious: https://github.com/graphql/graphql-js/commit/1e46389a1970bb3609f33029cfcf0a41f65ad999#diff-6738ab51d9c26ea2d4b271f22bb521b267a1e290755dc4cf5da78fcf86facb6c